File: retina/python/padovan.py
# See https://en.wikipedia.org/wiki/Padovan_sequence
# Original code from https://www.geeksforgeeks.org/padovan-sequence/

def padovan(n):
    pPrevPrev, pPrev, pCurr, pNext = 1, 1, 1, 1
    for i in range(3, n+1):
        pNext = pPrevPrev + pPrev
        pPrevPrev = pPrev
        pPrev = pCurr
        pCurr = pNext
    return pNext;

if __name__ == "__main__":
    cases = [
        "padovan(0)",
        "padovan(1)",
        "padovan(2)",
        "padovan(3)",
        "padovan(4)",
        "padovan(5)",
        "padovan(6)",
        "padovan(91)",
        "padovan(283)",
        "padovan(3674)"
    ]

    for c in cases:
        print('[] :python-answer """%s = %s""".' % (c, eval(c)))

File: reasoning/heisenberg/padovan.pl
% See https://en.wikipedia.org/wiki/Padovan_sequence

padovan(A,B) :-
    padovan(A,0,1,1,B).

padovan(0,A,_,_,A).
padovan(1,_,A,_,A).
padovan(2,_,_,A,A).
padovan(A,B,C,D,E) :-
    A > 2,
    F is A-1,
    G is B+C,
    padovan(F,C,D,G,E).

plastic_ratio(A,B) :-
    padovan(A,C),
    D is A+1,
    padovan(D,E),
    B is E/C.

% query implies goal
padovan(1,_ANSWER) => goal.
padovan(2,_ANSWER) => goal.
padovan(3,_ANSWER) => goal.
padovan(4,_ANSWER) => goal.
padovan(5,_ANSWER) => goal.
padovan(6,_ANSWER) => goal.
padovan(7,_ANSWER) => goal.
padovan(8,_ANSWER) => goal.
padovan(9,_ANSWER) => goal.
padovan(10,_ANSWER) => goal.
padovan(11,_ANSWER) => goal.
padovan(12,_ANSWER) => goal.
padovan(13,_ANSWER) => goal.
padovan(14,_ANSWER) => goal.
padovan(15,_ANSWER) => goal.
padovan(16,_ANSWER) => goal.
padovan(17,_ANSWER) => goal.
padovan(18,_ANSWER) => goal.
padovan(19,_ANSWER) => goal.
padovan(20,_ANSWER) => goal.
padovan(91,_ANSWER) => goal.
padovan(283,_ANSWER) => goal.
padovan(3674,_ANSWER) => goal.
plastic_ratio(1,_ANSWER) => goal.
plastic_ratio(10,_ANSWER) => goal.
plastic_ratio(100,_ANSWER) => goal.
plastic_ratio(1000,_ANSWER) => goal.
